### ISSUE TITLE
Crafting Cost changes w/ and w/o own materials

### ISSUE THREAD

**Issue by AkaTenshi** [gw2efficiency/issues#1305]
Hello,
I looked into crafting the minstrel and found that the total cost displayed changes depending if I choose with own materials or not.
![Screenshot from 2020-05-10 17-20-25](https://user-images.githubusercontent.com/6298910/81503749-96007580-92e5-11ea-8873-691b0c7f050e.png)
![Screenshot from 2020-05-10 17-20-39](https://user-images.githubusercontent.com/6298910/81503750-9731a280-92e5-11ea-9511-16bb4ccb0d98.png)
I can switch the dropdown back and forth and the values always end up like this. It's also been consistent across browser restarts yesterday and today. Same thing happens with the incinerator and the flameseeker prophecies for example even if to a lesser degree.
Also none of the 2 prices matches other sites like gw2bltc.com exactly. Though that may very well just be the timing of the last update of the dataset and not an actual bug.
In any case it's confusing and hard to tell how much it will actually cost to craft the item in total.

**Comment by queicherius** [comment]
> I looked into crafting the minstrel and found that the total cost displayed changes depending if I choose with own materials or not.

When you choose "own materials" it show the total cost **without** the materials you already have, so you can see how much you are missing. The total cost to craft it is the one where "Use Own Materials" is set to "No". Also see #224.

> Also none of the 2 prices matches other sites like gw2bltc.com exactly. Though that may very well just be the timing of the last update of the dataset and not an actual bug.

Yea, our update schedule is likely different - the "Use Own Materials: No" should be similar.

**Comment by AkaTenshi** [comment]
> When you choose "own materials: yes" it show the total cost without the price of the materials you already have, so you can see how much you are missing.
I am well aware of that. But I wasn't talking about the remaining cost but the total cost.
> The total cost to craft it is the one where "Use Own Materials" is set to "No".
Indeed. But is it also displayed below the remaining cost if you set "Use Own Materials" to "Yes" for reference.
Let me try to explain again:
In the first screenshot (w/o own mats) the total cost it is 1130g. In the second screenshot (w/ own mats) it's 1296g in total. The total cost to craft an item shouldn't change based on already owning a few of the mats.
For the vast majority of items the total cost of both modes are identical as expected. But for the examples I posted there is a difference which shouldn't be there.

**Comment by queicherius** [comment]
Oh! I get your question now, sorry for the confusion.

The reason for the difference is that when you **do not** choose "own materials" it calculates the best possible path for the item, with the prices of all items. When you **do** choose "own materials" it calculates the best possible path for the item, assuming that all materials you own are "free", which means that the "best possible path" might be different.

As an example:

- Own materials off: The cheapest way to craft is buying Mithril Ore, and crafting them into Mithril Ingots
- Own materials on: You already have Mithril Ingots, so it's using them. But the total theoretical cost to craft **without** your own material with this tree path would be higher (since you would need to buy Ingots directly, which is more expensive).

**Comment by AkaTenshi** [comment]
Ah, that makes sense. Did think about this briefly but wasn't sure if it was intended this way and got confused. Thanks for the clarification^^
